effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -675,50 +688,106 @@ object Namer extends Phase[Parsed, NameResolved] {
+          Context.info(pretty"Did you want to write a boxed type with effects, `${tpe.sourceOf} / ${effects.sourceOf} at ${capt.sourceOf}`?")
+        case source.Effectful(innerTpe, eff) =>
+          if isParam then Context.info(pretty"Did you mean to use braces and a function type `() => ${innerTpe.sourceOf} / ${eff.sourceOf}`?")
+          Context.info(pretty"Did you mean to use a first-class, boxed type `() => ${innerTpe.sourceOf} at {} / ${eff.sourceOf}`?")
+        case _ => ()
+
+      // Dummy value type in order to aggregate more errors (see #947)
+      ValueTypeApp(ErrorValueType(), Nil)
   }
+  def resolveValueType(tpe: source.ValueType)(using Context): ValueType = resolveValueType(tpe, isParam = false)
 
-  def resolve(tpe: source.BlockType)(using Context): BlockType = resolvingType(tpe) {
+  def resolveBlockType(tpe: source.BlockType, isParam: Boolean = false)(using Context): BlockType = resolvingType(tpe) {
     case t: source.FunctionType  => resolve(t)
     case t: source.BlockTypeTree => t.eff
-    case t: source.BlockTypeRef => resolve(t)
+    case t: source.TypeRef => resolveBlockRef(t)
+    case other =>
+      Context.error(pretty"Expected block type, but got ${describeType(other)}.")
+      other match
+        case source.BoxedType(innerTpe, eff) =>
+          if isParam then Context.info(pretty"Did you mean to use parentheses in order to receive a value type ${other.sourceOf}?")
+          Context.info(pretty"Did you mean to use the block type ${innerTpe.sourceOf} without 'at ${eff.sourceOf}'?")
+        case source.Effectful(source.FunctionType(tparams, vparams, bparams, result, funEffects), effects) =>
+          val combinedEffects = funEffects.effs.toSet ++ effects.effs.toSet
+          val prettyEffects = combinedEffects.toList.map { _.sourceOf } match
+            case List(eff) => eff
+            case Nil => "{}"
+            case many => many.mkString("{", ", ", "}")
+          // TODO(jiribenes, 2025-04-22): `Effects` seem to have bad position information. Why?!
+          Context.info(pretty"A function type cannot have multiple effect sets, did you mean to use `/ ${prettyEffects}` instead of `/ ${funEffects} / ${effects}`?")
+        case source.Effectful(source.BoxedType(tpe @ source.FunctionType(tparams, vparams, bparams, result, funEffects), capt), effects) =>
+          Context.info(pretty"Did you want to write a boxed type with effects, `${tpe.sourceOf} / ${effects.sourceOf} at ${capt.sourceOf}`?")
+        case source.Effectful(innerTpe, effs) =>
+          // NOTE: We could use `isParam` to write a more precise message, but what exactly would it be?
+          Context.info(pretty"Did you mean to use a function type () => ${innerTpe.sourceOf} / ${effs.sourceOf}?")
+        case _ => ()
+
+      // Dummy interface type in order to aggregate more errors (see #947)
+      InterfaceType(ErrorBlockType(), Nil)
```

Comment:
I am not entirely happy with the constructor here. What is the tradeoff of introducing it as a BlockType (and ValueType respectively)?

Reply:
In at least one case, we need to make it an interface type (which is a case class only!), hence the block constructor contained within it. We can reuse it for the block type case too.
Then the value constructor is defined for symmetry for value types.

Reply:
Ah right... we use it for effects!

This is super annoying. What would you do if we were to also implement error resilient type checking?

Reply:
My intuition is that if we were to represent things in untyped rows like `Symbol := (SymbolKind, Name, ...)`, then I'd have a `SymbolKind.Error` that can appear anywhere.
But I can't think of a good encoding of a "every symbol can be bogus" without devolving into `isNull`-style checks everywhere?

Reply:
One potential solution is to replace every bad expression with a hole that acts like a unification variable in Typer ala Agda.

Reply:
Yes, exactly: if we could, I'd replace the bad type with a brand new unification variable for a value type / block type / effect.